src/core/properties/multifileproperty.cpp

```diff
@@ -118,195 +140,144 @@ void MultiFileProperty::serialize(Serializer& s) const {
-
-AcceptMode MultiFileProperty::getAcceptMode() const { return acceptMode_; }
-
-void MultiFileProperty::setFileMode(FileMode mode) { fileMode_ = mode; }
-
-FileMode MultiFileProperty::getFileMode() const { return fileMode_; }
-
-void MultiFileProperty::setContentType(std::string_view contentType) { contentType_ = contentType; }
-
-std::string MultiFileProperty::getContentType() const { return contentType_; }
-
 void MultiFileProperty::requestFile() {
     for (auto widget : getWidgets()) {
-        if (auto filerequestable = dynamic_cast<FileRequestable*>(widget)) {
-            if (filerequestable->requestFile()) return;
+        if (auto fileRequester = dynamic_cast<FileRequestable*>(widget)) {
+            fileRequester->requestFile();
+            return;
         }
     }
-    if (getWidgets().empty()) {
-        // Currently, the only difference between using the widget (Qt) and the FileDialog directly
-        // is that the Qt widget remembers the previously used directory
-        auto fileDialog = util::dynamic_unique_ptr_cast<FileDialog>(
-            InviwoApplication::getPtr()->getDialogFactory()->create("FileDialog"));
-        if (!fileDialog) {
-            throw Exception(
-                "Failed to create a FileDialog. Add one to the InviwoApplication::DialogFactory",
-                IVW_CONTEXT);
-        }
-
-        // Setup Extensions
-        std::vector<FileExtension> filters = this->getNameFilters();
-        fileDialog->addExtensions(filters);
-
-        fileDialog->setCurrentFile(get().empty() ? "" : get().front());
-        fileDialog->setTitle(getDisplayName());
-        fileDialog->setAcceptMode(getAcceptMode());
-        fileDialog->setFileMode(getFileMode());
-
-        auto ext = getSelectedExtension();
-        if (!ext.empty()) fileDialog->setSelectedExtension(ext);
+    // No FileRequestable widget found, use the factory.
+    // Currently, the only difference between using the widget (Qt) and the FileDialog
+    // directly is that the Qt widget remembers the previously used directory
+    auto fileDialog =
+        createFileDialog(getDisplayName(), front() ? *front() : std::filesystem::path{});
+    if (fileDialog->show()) {
+        set(fileDialog->getSelectedFiles(), fileDialog->getSelectedFileExtension());
+    }
+}
 
-        if (fileDialog->show()) {
-            setSelectedExtension(fileDialog->getSelectedFileExtension());
-            set(fileDialog->getSelectedFiles());
-        }
+MultiFileProperty& MultiFileProperty::setCurrentStateAsDefault() {
+    Property::setCurrentStateAsDefault();
+    FileBase::setAsDefault();
+    files_.setAsDefault();
+    return *this;
+}
+MultiFileProperty& MultiFileProperty::setDefault(const std::vector<std::filesystem::path>& value) {
+    files_.defaultValue = value;
+    return *this;
+}
+MultiFileProperty& MultiFileProperty::resetToDefaultState() {
+    if (files_.reset() || FileBase::reset()) {
```

Comment:
Also here

Reply:
that was somewhat annoying... :/ liked not having to same the modified state in some variable...

Reply:
think I fixes all of them